eng/pipelines/templates/stages/archetype-java-release-batch.yml

```diff
@@ -0,0 +1,497 @@
+parameters:
+  - name: DependsOn
+    type: object
+  - name: ServiceDirectory
+    type: string
+  - name: SDKType
+    type: string
+  - name: Artifacts
+    type: object
+    default: []
+  - name: TestPipeline
+    type: boolean
+    default: false
+  - name: ArtifactName
+    type: string
+    default: 'not-specified'
+  - name: TargetDocRepoName
+    type: string
+    default: ''
+  - name: EnableIntegrationStage
+    type: boolean
+    default: true
+
+stages:
+  # The signing stage is responsible for submitting binaries to ESRP for our official signing
+  # where appropriate and also meeting any other signing requirements for particular artifacts,
+  # in this case we do GPG signing in order to publish to Maven Central. At the moment signing
+  # is protected by an approval check but this may be removed in the future.
+  - stage: Signing
+    dependsOn: ${{parameters.DependsOn}}
+    jobs:
+      - deployment: SignPackage
+        environment: esrp
+        timeoutInMinutes: 20
+        variables:
+          - template: /eng/pipelines/templates/variables/globals.yml
+        pool:
+          name: azsdk-pool-mms-ubuntu-2004-general
+          vmImage: MMSUbuntu20.04
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+                - checkout: none
+
+                - download: current
+                  displayName: 'Download Artifact: ${{parameters.ArtifactName}}'
+                  artifact: ${{parameters.ArtifactName}}
+
+                - template: tools/java-esrp-signing/java-esrp-signing.yml@azure-sdk-build-tools
+                  parameters:
+                    Artifacts: ${{parameters.Artifacts}}
+                    ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}
+
+                - publish: $(Pipeline.Workspace)/${{parameters.ArtifactName}}
+                  artifact: ${{parameters.ArtifactName}}-signed
+                  displayName: 'Store signed packages in ${{parameters.ArtifactName}}-signed artifact'
+
+  # We generate two interdependent stages for each artifact listed in the ci.yml file, creates the release
+  # in GitHub. The Release stage publishes to Maven Central. Both stages require approval since they
+  # effectively burn the version number. For testing of packages prior to burning the version number -
+  # the Validation step below publishes a package to a "burner" feed which is cleaned up after the
+  # pipeline completes.
+  - ${{if and(in(variables['Build.Reason'], 'Manual', ''), eq(variables['System.TeamProject'], 'internal'))}}:
+    - stage:
+      variables:
+        numArtifacts: ${{ length(parameters.Artifacts) }}
+      displayName: 'Releasing: ${{ variables.numArtifacts }} libraries'
```

Comment:
I don't expect that this will work. You can probably directly include ` length(parameters.Artifacts)` here though.

Reply:
@weshaggard this [does actually work](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1517042&view=results). Initially, I wasn't sure if it needed double curly braces or single square braces until I'd tested it.

Reply:
Interesting, not what I would have expected either.

Reply:
I changed it to displayName: 'Releasing: ${{ length(parameters.Artifacts) }} libraries' which also works just fine.